### PR TITLE
Metricbeat fails to build linux/mips* targets on linux/amd64 hosts

### PR DESCRIPTION
**Pull request by joereith**
- Bug

Changed to allow linux/mips* cross compilation on linux/amd64 hosts.

### CODE REVIEW

File: libbeat/metric/system/cgroup/cgv2/io_helper_linux.go [hunk 1]
```diff
@@ -48,13 +51,14 @@ func fetchDeviceName(major, minor uint64) (bool, string, error) {
...
+
 		// do some bitmapping to extract the major and minor device values
 		// The odd duplicated logic here is to deal with 32 and 64 bit values.
 		// see bits/sysmacros.h
-		curMajor := ((devID & 0xfffff00000000000) >> 32) | ((devID & 0x00000000000fff00) >> 8)
-		curMinor := ((devID & 0x00000000000000ff) >> 0) | ((devID & 0x00000ffffff00000) >> 12)
-		if curMajor == major && curMinor == minor {
+		curMajor = ((devID & 0xfffff00000000000) >> 32) | ((devID & 0x00000000000fff00) >> 8)
+		curMinor = ((devID & 0x00000000000000ff) >> 0) | ((devID & 0x00000ffffff00000) >> 12)
+		if uint64(curMajor) == uint64(major) && uint64(curMinor) == uint64(minor) {
```
Comment (fearful-symmetry) [inline libbeat/metric/system/cgroup/cgv2/io_helper_linux.go:61]: I'm not sure these explicit casts to `uint64()` are necessary here?

Comment (joereith) [reply libbeat/metric/system/cgroup/cgv2/io_helper_linux.go:61]: I think your correct.

**Comment by fearful-symmetry** [comment]
@joereith can you make those changes to remove the unnecessary casts, and also merge from master to fix CI?

**Comment by joereith** [comment]
@fearful-symmetry Just submitted the requested changes.

**Review by fearful-symmetry — APPROVED** [review]
LGTM. We might have to re-merge against master to fix CI.

**Comment by joereith** [comment]
@fearful-symmetry Did this pull request wind up making it through your CI/CD pipeline? I don't usually deal with complex build systems in git so I am not entirely familiar with the process, I apologize.

**Comment by fearful-symmetry** [comment]
@joereith it might be easier if you do it, but you need to re-merge the upstream changes from `elastic/beats`. Depending on how you have the git repo setup, it should be something like `git fetch upstream && git merge upstream/master`

**Comment by joereith** [comment]
I tried and i get an error stating:
"beats$ git fetch upstream && git merge upstream/master
fatal: 'upstream' does not appear to be a git repository
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
"

**Comment by fearful-symmetry** [comment]
Ah, there we go.

**Pull request merged**